integration/pvtdata/implicit_coll_test.go

```diff
@@ -164,3 +175,18 @@ func readImplicitCollection(n *nwo.Network, peer *nwo.Peer, chaincodeName string
 	}
 	queryChaincode(n, peer, command, expectedMsg, expectSuccess)
 }
+
+func discoverPeers(n *nwo.Network, peer *nwo.Peer, channelID string, retries int, retryInterval time.Duration) []nwo.DiscoveredPeer {
+	// retrt if discovery service does not return expected number peers
```

Comment:
IMO, two comments in this function are just distractions. It does provide new information. From the code, it is evident that we retry until all peers are discovered. 

Please refer to the comment section here -- https://docs.google.com/document/d/1mU4S4U1dDpBk9AA180RMRB7g3OH887nqbcEPGD3px3Q/edit?usp=sharing

Reply:
From the same doc it also says "Comments should provide reasoning". I'd suggest:
- put the comment above the function
- include reasoning in the comment

e.g.
// discoverPeers validates that all peers are discovered within the number of requested retries, to ensure that gossip has had a chance to initialize.

Reply:
This Function is straight-forward and I don't see a strong reason for having a comment. I am okay with having the comment above the function definition (though it is necessary only for the exported function as per golinters).

> Comments always lie and provide misinformation. This is because the comment won’t get evolved with code and it would be missed out unintentionally.

**Proof:** https://github.com/hyperledger/fabric/blob/master/core/ledger/pvtdatastorage/store.go#L21 You know what is wrong with the comment. It will be fixed in the refactoring. 

> Comments should provide reasoning

Here are some examples: 

1. https://github.com/hyperledger/fabric/blob/master/core/ledger/kvledger/kv_ledger.go#L422 
2. https://github.com/hyperledger/fabric/blob/master/core/ledger/kvledger/txmgmt/txmgr/lockbasedtxmgr/lockbased_txmgr.go#L140 
3. https://github.com/hyperledger/fabric/blob/master/core/ledger/kvledger/txmgmt/pvtstatepurgemgmt/purge_mgr.go#L140
4. https://github.com/hyperledger/fabric/blob/master/core/ledger/kvledger/txmgmt/pvtstatepurgemgmt/purge_mgr.go#L112
5. https://github.com/hyperledger/fabric/blob/master/core/ledger/kvledger/txmgmt/txmgr/lockbasedtxmgr/lockbased_txmgr.go#L109
6. https://github.com/hyperledger/fabric/blob/master/core/ledger/kvledger/txmgmt/txmgr/lockbasedtxmgr/lockbased_txmgr.go#L223

Without these comments, @manish-sethi and I would struggle to know why we did what we did.

Reply:
@denyeart @cendhu, thank you. Agree the function is straight-forward. Renamed the function to discoverAllPeers and removed the comment.